src/ngx_http_lua_shdict.c

```diff
@@ -513,10 +556,12 @@ ngx_http_lua_shdict_get_helper(lua_State *L, int get_stale)
 
     rc = ngx_http_lua_shdict_lookup(zone, hash, key.data, key.len, &sd);
 
+    ctx->lock_downgrade(ctx);
```

Comment:
Why downgrade the lock when we are still holding a write lock here? It looks wrong to me.